lib/authz/permissions_test.go

```diff
@@ -917,39 +901,22 @@ func TestAuthorizeWithVerbs(t *testing.T) {
 			}),
 			errAssertion: require.NoError,
 		},
-		{
-			name: "regular auth with verbs",
-			delegate: AuthorizerFunc(func(ctx context.Context) (*Context, error) {
-				accessChecker, err := services.NewAccessChecker(&services.AccessInfo{
-					Roles: []string{"test"},
-				}, "test-cluster", accessService)
-				require.NoError(t, err)
-				return &Context{
-					Checker: accessChecker,
-				}, nil
-			}),
-			kind:         types.KindUser,
-			verbs:        []string{types.VerbRead},
-			errAssertion: require.NoError,
-		},
 		{
 			name: "connection problem",
 			delegate: AuthorizerFunc(func(ctx context.Context) (*Context, error) {
-				return nil, trace.ConnectionProblem(errors.New("err msg"), "err msg")
+				return nil, connectionProblemErr
 			}),
 			errAssertion: func(t require.TestingT, err error, _ ...interface{}) {
-				require.True(t, trace.IsConnectionProblem(err))
-				require.Equal(t, "failed to connect to the database", err.Error())
+				require.ErrorIs(t, err, trace.ConnectionProblem(connectionProblemErr, "failed to connect to the database"))
```

Comment:
Exact comparison of trace errors is often brittle, I've had to refactor assertions like this more than once. Could we use a more resilient assertion? (For example, a combo of ErrorAs and ErrorContains.)

Same for others.